### DISCUSSION TITLE
Why is "guessed "dir" filetype based on contents" a warning?

### DISCUSSION THREAD

**Discussion by cool-RR** [saulpw/visidata#2072] [Q&A]
I'm moving the discussion from #2065 to here.

I wrote: 

> Is the "guessed "dir" filetype based on contents" message really helpful? Starting vd on a folder is the 99% use case for me. I don't need VisiData telling me that it was just scratching its head every time I open it.

@anjakefala wrote:

> There is a `git` filetype, which is why that guessing information is there. It's to let people know if it was opened as a git repo or dir repo. That has bitten me once before.

Okay, but why is it a warning though? My general understanding of warnings, not specifically with VisiData, is that they mean "you did something slightly wrong" ([relevant standup bit](https://www.youtube.com/watch?v=sQyzZulKAag).) When a warning becomes part of everyday usage people just become numb to warnings and ignore them, which is more dangerous.

Interestingly, the warning doesn't show up when you run `vd` but only when you run `vd some_folder`, even though the former opens a folder as well.

**Comment by saulpw** [comment]
> Interestingly, the warning doesn't show up when you run vd but only when you run vd some_folder, even though the former opens a folder as well.

This is because, in the former case, we know we're opening a folder, so we specify the filetype explicitly.  You can get the same effect by using `vd -f dir some_folder`.
